### ISSUE TITLE
Each Process requires GPU memory in TensorFlow 1.13.1

### ISSUE THREAD

**Issue by mx8435** [horovod/horovod#876]
**Environment:**
1. Framework: (TensorFlow, Keras, PyTorch, MXNet) Tensorflow
2. Framework version: 1.13.1
3. Horovod version: 0.6.0
4. MPI version: 4.0.0
5. CUDA version: 10.0
6. NCCL version: 2.4.2
7. Python version: 3.6.5
8. OS and version: OSX

**Your question:**
Hi @alsrgv, I test `np=8` on localhost, but I find that each process occupy a gpu memoy. Is my openmpi install failed? Or NCCL install failed?

`HOROVOD_TIMELINE=/tmp/timeline.json HOROVOD_GPU_ALLREDUCE=NCCL LD_LIBRARY_PATH=/usr/local/cuda-10.0/lib64:/usr/local/cuda-10.0/extras/CUPTI/lib64:/usr/local/nccl_2.4.2/lib  mpirun --allow-run-as-root -np 8  -H localhost:8  -bind-to none -map-by slot  -x NCCL_DEBUG=INFO -x LD_LIBRARY_PATH  -x PATH  -mca pml ob1 -mca btl ^openib  python main.py`

![image](https://user-images.githubusercontent.com/8417466/53690252-02a6e880-3da2-11e9-989d-e31c34588014.png)

**Comment by abditag2** [comment]
Each horovod process (or as we call it a rank) runs on one GPU. The number of processes that you specify must match the total number of GPUs you have on all your nodes.

**Comment by mx8435** [comment]
> GPU

Thanks for your reply. I've checked the number of GPUs (8) and the `np` option (np=8), it's match.  Is the openmpi or nccl  not install correct?

**Comment by tgaddair** [comment]
Hey @mx8435, is the issue here that every worker process is taking some memory from every GPU?

If so, did you remember to pin each process to a unique GPU?

```
config = tf.ConfigProto()
config.gpu_options.allow_growth = True
config.gpu_options.visible_device_list = str(hvd.local_rank())
```

Also, are you running with XLA enabled? XLA may cause issues with GPU memory being allocated across all processes.

**Comment by mx8435** [comment]
> Hey @mx8435, is the issue here that every worker process is taking some memory from every GPU?
> 
> If so, did you remember to pin each process to a unique GPU?
> 
> ```
> config = tf.ConfigProto()
> config.gpu_options.allow_growth = True
> config.gpu_options.visible_device_list = str(hvd.local_rank())
> ```
> Also, are you running with XLA enabled? XLA may cause issues with GPU memory being allocated across all processes.

XLA is not  used in my program and `config.gpu_options.visible_device_list = str(hvd.local_rank())` has been set.

**Comment by xajxiang** [comment]
@mx8435 do you find the solution yet?

**Comment by tgaddair** [comment]
@mx8435 does this problem also occur when you run standard Horovod examples?  For instance `tensorflow_synthetic_benchmark.py` or `tensorflow_mnist.py`?

**Comment by alsrgv** [comment]
@mx8435, @xajxiang, this is [a known issue](https://github.com/tensorflow/tensorflow/pull/24461#issuecomment-467235703) in TF 1.13.1.  Unfortunately, the fix did not make it into TF 1.13.x branch.  You can either downgrade to TF 1.12.0 or try out `tf-nightly-gpu`.

**Comment by xajxiang** [comment]
@alsrgv  but TF1.12.0 only support cuda 9.0. I see the issue releated a known issue in TF 1.13.1.

If I can close the XLA feature in 1.13.X,  then I can solve the problem

**Comment by alsrgv** [comment]
@xajxiang, if you have to continue using CUDA 10, you can try `tf-nightly-gpu`. Alternatively, you can try to [build TF from source](https://www.tensorflow.org/install/source) and disable XLA.  I’m not aware of a way to disable XLA without rebuilding TF 1.13.

**Comment by xajxiang** [comment]
@alsrgv thanks a lot,   I will try to build TF from soure and disable XLA. the solution to disable XLA is through  marco definiton or some thing?

**Comment by alsrgv** [comment]
@xajxiang, the configuration script will ask whether you want XLA enabled and you can say “N”.

**Comment by maxhgerlach** [comment]
Is this fixed in the upcoming TensorFlow 1.14?

**Comment by alsrgv** [comment]
@maxhgerlach, yes, it is.
